code/modules/mining/mine_turf_types.dm

```diff
@@ -22,6 +22,38 @@
 		underlay_appearance.icon_state += "[rand(0,12)]"
 	return TRUE
 
+/turf/simulated/lava/Entered(atom/movable/AM, atom/oldloc)
+	if(istype(AM, /mob/living))
+		var/mob/living/L = AM
+		if(locate(/obj/structure/lattice/catwalk, src))	//should be safe to walk upon
+			return 1
+		if(!istype(oldloc,/turf/simulated/lava))
+			to_chat(L, "<span class='warning'>You are covered by fire and heat from entering \the [src]!</span>")
+		if(ishuman(L))
```

Comment:
Did some lookings. Fire code is written at mob/living level, so this can be generalized a whole bunch. This should be all you need:
```dm
if (L.flying)
	return 1

L.adjust_fire_stacks(5) // or any number really.
L.IgniteMob()
```

The burning API will take care of adding temperature to the target itself, which will take care of adding fire damage in turn. So this is all you need.